specification/migrate/resource-manager/Microsoft.Migrate/stable/2019-05-01/migrate.json

```diff
@@ -0,0 +1,3778 @@
+      "description": "An error response from the Azure Migrate service."
+    },
+    "CollectorBodyAgentSpnProperties": {
+      "type": "object",
+      "properties": {
+        "authority": {
+          "description": "AAD Authority URL which was used to request the token for the service principal.",
+          "type": "string"
+        },
+        "applicationId": {
+          "description": "Application/client Id for the service principal with which the on-premise management/data plane components would communicate with our Azure services.",
+          "type": "string"
+        },
+        "audience": {
+          "description": "Intended audience for the service principal.",
+          "type": "string"
+        },
+        "objectId": {
+          "description": "Object Id of the service principal with which the on-premise management/data plane components would communicate with our Azure services.",
+          "type": "string"
+        },
+        "tenantId": {
+          "description": "Tenant Id for the service principal with which the on-premise management/data plane components would communicate with our Azure services.",
+          "type": "string"
+        }
+      }
+    },
+    "HyperVCollector": {
+      "type": "object",
+      "properties": {
+        "eTag": {
+          "type": "string"
+        },
+        "id": {
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/CollectorProperties"
+        }
+      }
+    },
+    "CollectorProperties": {
+      "type": "object",
+      "properties": {
+        "discoverySiteId": {
+          "type": "string"
+        },
+        "createdTimestamp": {
+          "type": "string",
+          "readOnly": true
+        },
+        "updatedTimestamp": {
+          "type": "string",
+          "readOnly": true
+        },
+        "agentProperties": {
+          "$ref": "#/definitions/CollectorAgentProperties"
+        }
+      }
+    },
+    "CollectorAgentProperties": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true
+        },
+        "version": {
+          "type": "string",
+          "readOnly": true
+        },
+        "lastHeartbeat": {
+          "type": "string",
+          "format": "date-time",
+          "x-nullable": true,
+          "readOnly": true
+        },
+        "spnDetails": {
+          "$ref": "#/definitions/CollectorBodyAgentSpnProperties"
+        }
+      }
+    },
+    "VMwareCollector": {
+      "type": "object",
+      "properties": {
+        "eTag": {
+          "type": "string"
+        },
+        "id": {
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "read
```

Comment:
This api is actually not paginated. Removing x-ms-pageable.

Reply:
That doesn't change the requirements on the format.  Since this is a collection GET it should wrap the result in a "value": [] array and if this is a singleton response represent the single assessmentOption as a resource with name "default".  Your other option is to model this as a POST action called listAssessmentOptions.  The POST action would let you keep the current response but you would need to map the authorization action from /action to /read in your manifest so users could still call it if the resource was locked

Reply:
Actually this is not a collection get this is a single get. I think the name is causing the confusion, renamed it appropriately.